DBZ-1988: Skip LSN timestamp query when using Oracle CDC jobs

jantpedraza opened the pull request:
Hi guys!

I have created this PR based on the below customer request ticket 

https://issues.redhat.com/browse/DBZ-1988

It would be really nice to have in place something similar.

Cheers,
@jantpedraza

gunnarmorling commented on debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java:
```diff
@@ -273,6 +274,17 @@ public static SnapshotIsolationMode parse(String value, String defaultValue) {
             .withDescription("The timezone of the server used to correctly shift the commit transaction timestamp on the client side"
                     + "Options include: Any valid Java ZoneId");
 
+    public static final Field SKIP_LSN_TIMESTAMP_QUERY = Field.create(DATABASE_CONFIG_PREFIX + SKIP_LSN_TIMESTAMP_QUERY_CONFIG_NAME)
```
The "database" prefix is only meant for pass-through options which are passed on to the JDBC driver. It's not needed here.

jantpedraza replied:
That was my first intention. I tried without the `database.` prefix, but I realized it was then filtered out at the moment `SqlServerConnection` was created:

```java
        final Configuration jdbcConfig = config.filter(
                x -> !(x.startsWith(DatabaseHistory.CONFIGURATION_FIELD_PREFIX_STRING) || x.equals(HistorizedRelationalDatabaseConnectorConfig.DATABASE_HISTORY.name())))
                .subset("database.", true);
        dataConnection = new SqlServerConnection(jdbcConfig);
```

Maybe I was doing something wrong, but as far as I remember, I had to include the prefix.

gunnarmorling replied:
Would be good to double-check. It shouldn't be needed.

gunnarmorling commented on debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java:
```diff
@@ -44,6 +44,7 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SqlServerConnectorConfig.class);
 
+    public static final String SKIP_LSN_TIMESTAMP_QUERY_CONFIG_NAME = "skip.lsn.timestamp.query";
```
I'd suggest to name this option more after the semantics of the timestamp. Perhaps "source.timestamp.mode" = {commit|processing}. I.e. we wouldn't expose technicalities like how the value is retrieved (e.g. this might change in a future SQL Server version where we could get the value without such query, but its semantics of being commit-LSN-based would still be the same). Also it'd open room for adding other strategies later on, if needed.

jantpedraza replied:
This makes all the sense to me. Honestly, I was not really happy with the property name I suggested since it is quite specific on how it was implemented, without having a clear semantic. I will use your suggestion! Thanks.

gunnarmorling commented on debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java:
```diff
@@ -273,6 +274,17 @@ public static SnapshotIsolationMode parse(String value, String defaultValue) {
+            .withImportance(Importance.LOW)
+            .withDescription("Skips the query which maps a commit LSN to a point in time when the commit happened." +
+                    "Options include:" +
+                    "'true', which means the LSN timestamp is defaulted to Instant.now()." +
```
Should be updated as per above. Also let's avoid exposing internals like the usage of `Instant`: "defaulted to the timestamp of processing this change event". Non-Java users wouldn't know what it is anyways.

jantpedraza replied:
Nice suggestion. Thanks!

gunnarmorling commented:
Sweet; just ping me if it's ready for another review round. Thanks!

gunnarmorling commented:
Hey @jantpedraza, just checking in on this one. It definitely will need a rebase. Thanks!

gunnarmorling commented:
Hi @jantpedraza; are you still planning to wrap this one up? No problem if not, just let us know so someone else can drive it home. Thanks!

jantpedraza commented:
Sorry @gunnarmorling, last week was a bit messy and I had no time for applying these changes. I will apply them and update this review :)

gunnarmorling commented:
Excellent, thank you so much! Please don't feel pressured, just trying to
get some PRs off the plate which have been around for a while now :)

>

jantpedraza commented:
Hi @gunnarmorling! I hope you are having a good weekend :)

Finally, I committed the required changes but I did it under a new PR (https://github.com/debezium/debezium/pull/1486). I don't know why I wasn't able to rebase from master my branch...

Anyway, here you have a summary of all these changes:
- Created a `SourceTimestampMode` enum to include all the possible options for the source timestamp field `ts_ms`.
- Removed the `database.` prefix from the config name. I would ask you, please, to make sure this is not required since the last time I tried I had to add it.
- Added the documentation in _sqlserver.adoc_.
- Changed the logic to return `null` instead of `Instant.now()`.

Nothing else, feel free to add any change should you think it is needed :)

Thanks Gunnar,

@jantpedraza
